memo_app.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'cgi'
+require 'json'
+
+set :environment, :production
+
+class MemoData
+  attr_accessor :memos, :id
+
+  def initialize
+    @memos = {}
+    @id = 0
+  end
+
+  def add_memo(title, content)
+    @memos[@id] = { title: title, content: content }
+    rewrite_file
+    @id += 1
+  end
+
+  def delete_memo(id)
+    @memos.delete(id)
+    rewrite_file
+  end
+
+  def rewrite_file
+    open('data.txt', 'w') do |f|
+      f.write @memos.to_json
+    end
+  end
+
+  def select_memo(params)
+    id = params[:id].to_i
+    title = CGI.escapeHTML(@memos.dig(id, :title))
+    content = CGI.escapeHTML(@memos.dig(id, :content))
+    [id, title, content]
+  end
+
+  def read_memo
+    loaded_data = File.read('data.txt')
+    return if loaded_data == ''
+
+    memos = JSON::Parser.new(loaded_data, symbolize_names: true).parse
+    @memos = memos.transform_keys { |k| k.to_s.to_i }
+    @id = @memos.keys[-1] + 1  # メモIDの最新を取得し、１送る
```

Comment:
ハッシュのキー（ID）を数値にする必要があるのは、足し算をして最新の ID を取るためですよね。
それ以外の場所で、キーが数値である必要はなさそうです。
最新の ID を取得する処理を `#add_memo` に移したので、ここではキーの変換を行わないようにしてみました。

そうするとこんな感じでシンプルな処理になります。
```suggestion
    loaded_data.empty? ? {} : JSON::Parser.new(loaded_data, symbolize_names: true).parse
```